crates/pgt_workspace/src/workspace/server/tree_sitter.rs

```diff
@@ -20,24 +19,34 @@ impl TreeSitterStore {
 
         TreeSitterStore {
             db: DashMap::new(),
-            parser: RwLock::new(parser),
+            parser: Mutex::new(parser),
```

Comment:
nice that you changed this back.
I think in general we can opt for simplicity over performance/concurrency optimizations – there'll only ever be very few users